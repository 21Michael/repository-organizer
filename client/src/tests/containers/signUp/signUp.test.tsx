import React from "react";
import { render, cleanup } from "@testing-library/react";
import SignUp from "../../../containers/signUp/signUp";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter } from "react-router-dom";
import store from "../../../storeReduxToolkit/configStore";
import { Provider } from "react-redux";
import "mutationobserver-shim";

export default describe("SignUp: ", () => {
  afterEach(() => {
    cleanup();
  });

  test("signUp", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
