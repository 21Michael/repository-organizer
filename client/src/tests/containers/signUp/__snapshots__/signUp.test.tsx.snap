// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp:  signUp 1`] = `
<div>
  <div
    class="wrapper"
  >
    <form
      class="form"
    >
      <h1
        class="title"
      >
         
        Create an account
         
      </h1>
      <div
        class="wrapper"
      >
        <label
          class="label"
        >
          User name
          :
          <div
            class="wrapper"
          >
            <input
              autocomplete="user-name"
              class="input"
              name="user"
              placeholder="Surname Name"
              type="text"
              value=""
            />
            <span
              class="input__inform"
            />
          </div>
        </label>
        <label
          class="label"
        >
          Email
          :
          <div
            class="wrapper"
          >
            <input
              autocomplete="email"
              class="input"
              name="email"
              placeholder="email@gmail.com"
              type="email"
              value=""
            />
            <span
              class="input__inform"
            />
          </div>
        </label>
        <label
          class="label"
        >
          Create a password
          :
          <div
            class="wrapper"
          >
            <input
              autocomplete="new-password"
              class="input"
              name="password"
              placeholder="********"
              type="password"
              value=""
            />
            <span
              class="input__inform"
            >
              2 uppercase, 3 lowercase, 2 digits, 1 symbol
            </span>
          </div>
        </label>
        <label
          class="label"
        >
          Confirm the password
          :
          <div
            class="wrapper"
          >
            <input
              autocomplete="new-password"
              class="input"
              name="passwordConfirm"
              placeholder="********"
              type="password"
              value=""
            />
            <span
              class="input__inform"
            />
          </div>
        </label>
      </div>
      <div
        class="wrapper"
      >
        <button
          class="button null"
          name="submit"
          type="submit"
        >
          Sign Up
        </button>
      </div>
    </form>
  </div>
</div>
`;
